gitlab/config.py

```diff
@@ -150,6 +151,16 @@ def __init__(
         except Exception:
             pass
 
+        for attr in ("job_token", "http_password", "private_token", "oauth_token"):
+            value = getattr(self, attr)
+            prefix = "lookup:"
+            if isinstance(value, str) and value.lower().strip().startswith(prefix):
+                helper = value[len(prefix) :].strip()
+                value = (
+                    subprocess.check_output(helper, shell=True).decode("utf-8").strip()
```

Comment:
I think we should at least warn the user in the docs about `shell=True` or link to https://docs.python.org/3/library/subprocess.html#security-considerations. I guess you added it here so people can use pipes etc.

Also I guess this opens the user to arbitrary commands executed on the machine if someone manages to put a config file with precedence or in `PYTHON_GITLAB_CFG` env var. @max-wittig WDYT about using `shell=True` and this 2nd point here, does it matter? :thinking:

Reply:
We could also rather run a simple command and leave it up to the user to create a shell script.  If I remember right, this is the behavior from docker.

Reply:
Removed the ``shell=True`` and instead allow only a path to the helper program.  I also updated the documentation.

Reply:
Thank you for the changes @klorenz and sorry for the delay! I'll be merging in a minute.

I really liked the convenience of your initial implementation, and I don't want to block this PR any longer so I just pushed a small change (I credited you as the author still) - I went for a compromise where `shell=False` is still used and recommend wrappers for any scripts that require piping etc. I hope this is ok with you. 
I also expanded the docs and wrapped the call in a try/except block for people to have a meaningful erorr message on the CLI. See https://github.com/python-gitlab/python-gitlab/pull/1359/commits/91ffb8e97e213d2f14340b952630875995ecedb2, let me know if you want to revisit this in a follow-up.